syft/serde/serde.py

```diff
@@ -438,20 +439,21 @@ def _simplify(obj: object) -> object:
         ValueError: if `move_this` or `in_front_of_that` are not both single ASCII
         characters.
     """
-    try:
-        # check to see if there is a simplifier
-        # for this type. If there is, run return
-        # the simplified object
-        current_type = type(obj)
-        result = (simplifiers[current_type][0], simplifiers[current_type][1](obj))
-        return result
-
-    except KeyError:
-        # if there is not a simplifier for this
-        # object, then the object is already a
-        # simple python object and we can just
-        # return it
-        return obj
+    # check to see if there is a simplifier
```

Comment:
Yes this would be impact serialization too severely. I think there are 2 options:
- Either just add VirtualWorkers to the list `OBJ_FORCE_FULL_SIMPLIFIER_AND_DETAILERS = [BaseWorker, VirtualWorker]` => very easy but need to be done for all workers
- do the logic in the failure step `except KeyError:`, ie in that case try to see if some inheritance would be satisfied, if so build dynamically a dict of inheritance (here {VirtualWorker: BaseWorker} and a registry of non-inherited types (like {int, float, str, ... }). This way, there is a constant number of inheritance test and you only pay of a lookup in those 2 structures

Reply:
Why so?
In that case option 2 will work, it makes the code not so readable, but would be fast

Reply:
Do Grid workers inherit from BaseWorker?

Reply:
In that case option 2 says:
try as usual, and if you get a KeyError try to see if there is an inheritance with a parent class that would have a simplifer/detailers.
Because you don't want to do this at each except (this op is costly), each time you see a type and do the inheritance check, you add it in a kind of cache, smthg like a dict {VirtualWorker: BaseWorker, etc} if an inheritance was found or a set {int, float, str, ... } for the types where no inheritance was found.